src/airmar/airmar_processor.py

```diff
@@ -1,84 +1,157 @@
 import math
-from src.airmar.nmeaparser.nmea_sentence import get_sentence_interface
+
+from threading import Lock
+
+from src.airmar.airmar_exceptions import InvalidIDException, UnsupportedIDException
+from src.utils.vec import Vec2
+
 
 class AirmarProcessor:
-    def __init__(self, broadcaster):
+    """ Processes raw data from airmar and publishes to the specified
+    broadcaster """
+
+    def __init__(self, broadcaster, parser):
+        """ Builds an airmar processor to process raw airmar data.
+
+        Keyword Arguments:
+        broadcaster -- data broadcast type
+        parser -- nmea parser to map raw data.
+
+        Returns:
+        A new airmar processor
+        """
+        self.parser = parser
         self.broadcaster = broadcaster
-        self.data = {
-            "wind speed apparent" : None,
-            "wind angle apparent" : None,
-            "wind speed true": None,
-            "wind angle true": None,
-            "boat latitude": None,
-            "boat longitude": None,
-            "boat heading" : None,
-            "boat speed": None
-        }
-        self.raw = dict()
+
+        self.data = {}
 
     def update_airmar_data(self, nmea):
+        """ Updates airmar data from given nmea fields. 
+        
+        Keyword Arguments:
+        nmea -- A list of fields parsed from an nmea sentence
+
+        Side effects:
+        Broadcasts updated processed data.
+
+        Raises:
+        InvalidIDException for invalid sentence ids.
+        UnsupportedIDException when program does not implement
+        interface to parse given nmea fields.
+        """
+        raw = {}
         sid = nmea[0]
-        if sid is None:
-            return
-        get_sentence_interface(sid).update_data(self.raw, nmea)
-
-        # TODO Update list with more interfaces
-        if sid in ["WIVWT", "WIVWR"]:
-            self._update_wind(sid)
-        elif sid in ["GPGGA"]:
-            self._update_boat_gps(sid)
-        elif sid in ["GPVTG"]:
-            self._update_boat_speed(sid)
-
-        self.broadcaster.update_data(data=self.data)
-        for key in self.data.keys():
-            self.broadcaster.read_data(key=key)
-
-    def _update_wind(self, sid):
-        speed_key = "wind speed apparent"
-        angle_key = "wind angle apparent"
-        if sid is "WIVWT":
-            # Update true wind values instead
-            speed_key = "wind speed true"
-            angle_key = "wind angle true"
-        # Set old values = new values if no old values
-        if self.data[angle_key] is None:
-            self.data[angle_key] = float(self.raw[sid]["wind_angle_degree"]) % 360
-        if self.data[speed_key] is None:
-            self.data[speed_key] = float(self.raw[sid]["wind_speed_mps"])
 
+        if self.parser.update_data(data=raw, fields=nmea) is None:
```

Comment:
if you are using raw as your way of getting updated data, why does the parser update_data still return data... you should decide whether you want to get updated data through a return statement in parser.update_data or through the mutability of the raw dict